[Kotlin-Spring] support to Spring boot3 & jakarta extension

parenko opened the pull request:
I have used #13620 PR then just merged with master and fixed conflicts.
Also changed spring boot version in POM and Gradle.

Synchronized with Java generator options.

Additionally run generate samples for Kotlin and run export documentation.

Executed successfully `mvn clean compile` on following samples:
* samples/server/petstore/kotlin-springboot-3
* samples/server/petstore/kotlin-springboot
* samples/server/petstore/kotlin-springboot-reactive
* samples/server/petstore/kotlin-springboot-springfox
* samples/server/petstore/kotlin-springboot-delegate
* samples/server/petstore/kotlin-springboot-source-swagger1
* samples/server/petstore/kotlin-springboot-modelMutable
* samples/server/petstore/kotlin-springboot-source-swagger2

To fix #13578

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh bin/configs/kotlin*
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

FYI @jimschubert (2017/09) [❤️](https://www.patreon.com/jimschubert), @dr4ke616 (2018/08) @karismann (2019/03) @Zomzog (2019/04) @andrewemery (2019/10) @4brunu (2019/11) @yutaka0m (2020/03)

wing328 commented:
@parenko is this ready for merge? if yes, please mark is as "ready for review"

parenko commented:
@wing328 it was, just wanted to make sure that all checks passed.

jlengrand commented on modules/openapi-generator/src/main/resources/kotlin-spring/libraries/spring-boot/buildGradle-sb3-Kts.mustache:
```diff
@@ -0,0 +1,57 @@
+    id("org.jetbrains.kotlin.jvm") version kotlinVersion
+    id("org.jetbrains.kotlin.plugin.jpa") version kotlinVersion
+    id("org.jetbrains.kotlin.plugin.spring") version kotlinVersion
+    id("org.springframework.boot") version "3.0.0-M5"
```
I see 3.0.2 is out already, should we not remove M5 here?

jlengrand commented on modules/openapi-generator/src/main/resources/kotlin-spring/libraries/spring-boot/buildGradle-sb3-Kts.mustache:
```diff
@@ -0,0 +1,57 @@
+
+dependencies {
+{{#reactive}}    val kotlinxCoroutinesVersion = "1.6.1"
+{{/reactive}}    implementation("org.jetbrains.kotlin:kotlin-stdlib-jdk8")
```
Nice to have implementation here, instead of compile like for the default template. compile doesn't exist any more for Gradle 7

parenko replied:
Should I change something?

jlengrand replied:
Nope, just praising you!

parenko replied:
kudos to @jayandran-Sampath

jlengrand commented:
I've tried to regenerate the sample using the added `SpringBoot3: "true"` option for the default spring boot sample and got it to compile and run without issues. 
Nice PR, with some indentation fixes as well.

By default currently, no Spring Boot 3 sample is generated given that it's behind a flag. I feel It'd be useful to have at last one generated so it becomes part of the main sample suite. What do you think? After all, it's a major version

jlengrand commented:
Also tested the unhappy sprinfox path, working as expected. Well done!

jlengrand commented:
Which open issue would this close BTW @parenko ?

parenko commented:
> Which open issue would this close BTW @parenko ?

#14369 13578

jlengrand commented:
@parenko can you give me your opinion on this comment : 

`By default currently, no Spring Boot 3 sample is generated given that it's behind a flag. I feel It'd be useful to have at last one generated so it becomes part of the main sample suite. What do you think? After all, it's a major version`.

I'm happy to do the commit.

parenko commented:
@jlengrand thank you for pointing this out to me. I think it is a good idea and useful.

I tried your suggestion. Let me know if you would have done it similarly?

parenko commented:
I used wrong option in `bin/configs/kotlin-spring-boot-3.yaml`, will change it soon...

jlengrand commented:
Hey @parenko. I went ahead and changed the option for you. I decided to stay consistent with the "use" keyword. If you merge https://github.com/parenko/openapi-generator/pull/1/files It should run fine.

I also considered doing something like "springBootVersion = 3", with 2 being the default but I guess it's better to merge that now and consider it later (when is 4.x coming out anyways :)).


As soon as https://github.com/parenko/openapi-generator/pull/1/files is merged, AFAIC it's good to go @wing328

jlengrand commented:
Great job overall, it's clean, does what it's supposed to and you went the extra mile to test. Much appreciated

parenko commented:
@jlengrand, I have changed the `SpringBoot3` option to `useSpringBoot3` as in Java generator.
Also changed importing with `{{javaxPackage}}`. Similar to #14505 or as done in Java generator with #14310.
Now all tests pass.

jlengrand commented:
Looks great! Let's merge this @wing328 ! Thanks for all the work @parenko
